// @lgtm-hq/turbo-themes/packages/core/dist/themes/css/syntax-generator.js
// SPDX-License-Identifier: MIT
// Syntax highlighting CSS generation
import { escapeCssId, sanitizeCssColor } from './helpers.js';
import { SYNTAX_CLASS_GROUPS } from './mappings.js';
/**
 * Generates syntax highlighting CSS for a theme flavor.
 */
export function generateSyntaxHighlightingCSS(flavor) {
    const { tokens } = flavor;
    const escapedId = escapeCssId(flavor.id);
    // Sanitize all color values to prevent CSS injection
    const syntaxColors = {
        fg: sanitizeCssColor(tokens.content.codeInline.fg),
        bg: sanitizeCssColor(tokens.content.codeInline.bg),
        comment: sanitizeCssColor(tokens.text.secondary),
        keyword: sanitizeCssColor(tokens.brand.primary),
        string: sanitizeCssColor(tokens.state.success),
        number: sanitizeCssColor(tokens.state.warning),
        title: sanitizeCssColor(tokens.state.info),
        attr: sanitizeCssColor(tokens.accent.link),
    };
    const selector = (classes) => classes.map((c) => `html[data-flavor='${escapedId}'] .highlight ${c}`).join(',\n');
    return `html[data-flavor='${escapedId}'] .highlight {
  background: ${syntaxColors.bg};
  color: ${syntaxColors.fg};
}
html[data-flavor='${escapedId}'] .highlight pre,
html[data-flavor='${escapedId}'] pre.highlight {
  background: transparent;
  color: ${syntaxColors.fg};
}
html[data-flavor='${escapedId}'] .highlight code {
  background: transparent;
  color: ${syntaxColors.fg};
}
${selector(SYNTAX_CLASS_GROUPS.comment)} {
  color: ${syntaxColors.comment};
}
${selector(SYNTAX_CLASS_GROUPS.keyword)} {
  color: ${syntaxColors.keyword};
}
${selector(SYNTAX_CLASS_GROUPS.string)} {
  color: ${syntaxColors.string};
}
${selector(SYNTAX_CLASS_GROUPS.number)} {
  color: ${syntaxColors.number};
}
${selector(SYNTAX_CLASS_GROUPS.attr)} {
  color: ${syntaxColors.attr};
}
${selector(SYNTAX_CLASS_GROUPS.title)} {
  color: ${syntaxColors.title};
}`;
}
