// @lgtm-hq/turbo-themes/packages/core/dist/tokens/index.js
// SPDX-License-Identifier: MIT
/**
 * Platform-agnostic design tokens export
 *
 * This module exports pure theme data (colors, typography, etc.) without
 * any DOM, CSS, or platform-specific code. Use this for:
 *
 * - React Native apps
 * - Flutter apps (via JSON export)
 * - iOS/Android native apps
 * - Any platform that needs theme colors
 *
 * For web apps, you can also import the CSS files directly:
 * import '@lgtm-hq/turbo-themes/css/catppuccin-mocha.css'
 */
export { THEME_IDS } from '../themes/theme-ids.js';
// Import tokens data (copied from dist/tokens.json during build)
import tokensData from '../themes/tokens.json' with { type: 'json' };
const tokens = tokensData;
/**
 * All available theme flavors as a flat array
 */
export const flavors = Object.values(tokens.themes).map((theme) => ({
    id: theme.id,
    label: theme.label,
    vendor: theme.vendor,
    appearance: theme.appearance,
    ...(theme.iconUrl !== undefined && { iconUrl: theme.iconUrl }),
    tokens: theme.tokens,
}));
/**
 * Theme flavors indexed by ID for quick lookup
 */
export const themesById = /*#__PURE__*/ Object.fromEntries(flavors.map((flavor) => [flavor.id, flavor]));
/**
 * All available theme packages (grouped by vendor)
 */
export const packages = /*#__PURE__*/ Object.fromEntries(Object.entries(tokens.byVendor).map(([vendorId, vendor]) => [
    vendorId,
    {
        id: vendorId,
        name: vendor.name,
        homepage: vendor.homepage,
        flavors: vendor.themes.map((themeId) => themesById[themeId]).filter(Boolean),
    },
]));
/**
 * Get a theme by ID
 * @param id - Theme identifier (e.g., 'catppuccin-mocha')
 * @returns The theme flavor or undefined if not found
 */
export function getTheme(id) {
    return themesById[id];
}
/**
 * Get theme tokens by ID (convenience function)
 * @param id - Theme identifier (e.g., 'catppuccin-mocha')
 * @returns The theme tokens or undefined if not found
 */
export function getTokens(id) {
    return themesById[id]?.tokens;
}
/**
 * Get all themes matching an appearance (light/dark)
 * @param appearance - 'light' or 'dark'
 * @returns Array of matching theme flavors
 */
export function getThemesByAppearance(appearance) {
    return flavors.filter((f) => f.appearance === appearance);
}
/**
 * Get all themes from a specific vendor
 * @param vendor - Vendor name (e.g., 'catppuccin', 'github')
 * @returns Array of matching theme flavors
 */
export function getThemesByVendor(vendor) {
    return flavors.filter((f) => f.vendor === vendor);
}
/**
 * List of all available theme IDs
 */
export const themeIds = /*#__PURE__*/ flavors.map((f) => f.id);
/**
 * List of all available vendors
 */
export const vendors = /*#__PURE__*/ [...new Set(flavors.map((f) => f.vendor))];
