// @lgtm-hq/turbo-themes/packages/css/dist/index.js
//#region src/syntax.ts
function generateSyntaxVarsFromTokens(tokens) {
	const lines = [];
	const add = (name, value) => {
		lines.push(`  --${CSS_VAR_PREFIX}-syntax-${name}: ${value};`);
	};
	add("fg", tokens.content?.codeBlock?.fg ?? tokens.text.primary);
	add("bg", tokens.content?.codeBlock?.bg ?? tokens.background.surface);
	add("comment", tokens.text.secondary);
	add("keyword", tokens.brand.primary);
	add("string", tokens.state.success);
	add("number", tokens.state.warning);
	add("function", tokens.state.info);
	add("type", tokens.brand.primary);
	add("variable", tokens.text.primary);
	add("operator", tokens.text.secondary);
	add("punctuation", tokens.text.secondary);
	add("tag", tokens.state.danger);
	add("attribute", tokens.state.info);
	add("value", tokens.state.success);
	add("error", tokens.state.danger);
	add("deleted", tokens.state.danger);
	add("inserted", tokens.state.success);
	return lines;
}
var CSS_SYNTAX = `/* Turbo Themes - Syntax Highlighting */

/* Astro Code component css-variables theme mapping */
:root {
  --astro-code-color-text: var(--turbo-syntax-fg, var(--turbo-text-primary));
  --astro-code-color-background: var(--turbo-syntax-bg, var(--turbo-code-block-bg));
  --astro-code-token-constant: var(--turbo-syntax-number);
  --astro-code-token-string: var(--turbo-syntax-string);
  --astro-code-token-comment: var(--turbo-syntax-comment);
  --astro-code-token-keyword: var(--turbo-syntax-keyword);
  --astro-code-token-parameter: var(--turbo-syntax-variable);
  --astro-code-token-function: var(--turbo-syntax-function);
  --astro-code-token-string-expression: var(--turbo-syntax-string);
  --astro-code-token-punctuation: var(--turbo-syntax-punctuation);
  --astro-code-token-link: var(--turbo-accent-link, var(--turbo-brand-primary));
  /* Additional Astro tokens */
  --astro-code-background: var(--turbo-syntax-bg, var(--turbo-code-block-bg));
  --astro-code-foreground: var(--turbo-syntax-fg, var(--turbo-text-primary));
}

/* Astro Code component styling */
.astro-code,
pre.astro-code {
  background-color: var(--turbo-syntax-bg, var(--turbo-code-block-bg)) !important;
  padding: var(--space-lg);
  border-radius: var(--radius-lg);
  overflow-x: auto;
  border: 1px solid var(--turbo-border-default);
}

.astro-code code {
  background-color: transparent !important;
  color: inherit;
}

/* Base highlight container - only applies to pre elements */
.highlight,
pre.highlight,
pre[class*="language-"] {
  background-color: var(--turbo-syntax-bg, var(--turbo-code-block-bg));
  color: var(--turbo-syntax-fg, var(--turbo-code-block-fg));
}

/* Code inside pre should have transparent background */
.highlight pre,
.highlight code,
pre code,
pre code[class*="language-"] {
  background-color: transparent;
  color: inherit;
}

/* Comments */
.highlight .c,
.highlight .cm,
.highlight .c1,
.highlight .cs,
.highlight .cp,
.highlight .cpf,
.token.comment,
.token.prolog,
.token.doctype,
.token.cdata,
.hljs-comment,
.hljs-quote {
  color: var(--turbo-syntax-comment);
  font-style: italic;
}

/* Keywords */
.highlight .k,
.highlight .kc,
.highlight .kd,
.highlight .kn,
.highlight .kp,
.highlight .kr,
.highlight .kt,
.highlight .kv,
.token.keyword,
.token.atrule,
.token.important,
.hljs-keyword,
.hljs-selector-tag,
.hljs-built_in,
.hljs-type {
  color: var(--turbo-syntax-keyword);
}

/* Strings */
.highlight .s,
.highlight .s1,
.highlight .s2,
.highlight .sb,
.highlight .sc,
.highlight .sd,
.highlight .se,
.highlight .sh,
.highlight .si,
.highlight .sr,
.highlight .ss,
.highlight .sx,
.token.string,
.token.char,
.token.regex,
.token.template-string,
.hljs-string,
.hljs-regexp {
  color: var(--turbo-syntax-string);
}

/* Numbers */
.highlight .m,
.highlight .mi,
.highlight .mf,
.highlight .mh,
.highlight .mo,
.highlight .mb,
.highlight .mx,
.highlight .il,
.token.number,
.token.boolean,
.token.constant,
.hljs-number,
.hljs-literal {
  color: var(--turbo-syntax-number);
}

/* Functions */
.highlight .nf,
.highlight .fm,
.highlight .nb,
.token.function,
.token.builtin,
.hljs-title.function_,
.hljs-built_in {
  color: var(--turbo-syntax-function);
}

/* Classes and types */
.highlight .nc,
.highlight .nn,
.highlight .no,
.highlight .nd,
.token.class-name,
.token.namespace,
.hljs-title.class_,
.hljs-class .hljs-title {
  color: var(--turbo-syntax-type);
}

/* Variables and identifiers */
.highlight .n,
.highlight .nv,
.highlight .vc,
.highlight .vg,
.highlight .vi,
.highlight .vm,
.token.variable,
.hljs-variable,
.hljs-template-variable {
  color: var(--turbo-syntax-variable);
}

/* Operators */
.highlight .o,
.highlight .ow,
.token.operator,
.hljs-operator {
  color: var(--turbo-syntax-operator);
}

/* Punctuation */
.highlight .p,
.token.punctuation,
.hljs-punctuation {
  color: var(--turbo-syntax-punctuation);
}

/* HTML/XML tags */
.highlight .nt,
.highlight .nx,
.token.tag,
.hljs-tag,
.hljs-name {
  color: var(--turbo-syntax-tag);
}

/* Attributes */
.highlight .na,
.highlight .atn,
.token.attr-name,
.hljs-attr {
  color: var(--turbo-syntax-attribute);
}

/* Attribute values */
.highlight .atv,
.token.attr-value,
.hljs-attr .hljs-string {
  color: var(--turbo-syntax-value);
}

/* Errors */
.highlight .err,
.highlight .gr,
.token.error,
.hljs-error {
  color: var(--turbo-syntax-error);
}

/* Diff: deletions */
.highlight .gd,
.token.deleted,
.hljs-deletion {
  color: var(--turbo-syntax-deleted);
  background-color: color-mix(in srgb, var(--turbo-syntax-deleted) 15%, transparent);
}

/* Diff: insertions */
.highlight .gi,
.token.inserted,
.hljs-addition {
  color: var(--turbo-syntax-inserted);
  background-color: color-mix(in srgb, var(--turbo-syntax-inserted) 15%, transparent);
}

/* Line numbers (if used) */
.highlight .lineno,
.line-numbers .line-numbers-rows,
.hljs-ln-numbers {
  color: var(--turbo-text-secondary);
  opacity: 0.5;
  user-select: none;
}

/* Selection within code */
.highlight ::selection,
pre[class*="language-"] ::selection,
code[class*="language-"] ::selection {
  background-color: var(--turbo-selection-bg);
  color: var(--turbo-selection-fg);
}
`;
function generateSyntaxCss(flavor) {
	const vars = generateSyntaxVarsFromTokens(flavor.tokens);
	return `/* Turbo Themes - Syntax Highlighting for ${flavor.label} */
[data-theme="${flavor.id}"] {
${vars.join("\n")}
}
`;
}
function generateSyntaxBaseCss() {
	return CSS_SYNTAX;
}
var token_mappings_default = {
	$schema: "./token-mappings.schema.json",
	$description: "Single source of truth for CSS variable to token path mappings",
	prefix: "turbo",
	coreMappings: [
		{
			"cssVar": "bg-base",
			"tokenPath": "background.base"
		},
		{
			"cssVar": "bg-surface",
			"tokenPath": "background.surface"
		},
		{
			"cssVar": "bg-overlay",
			"tokenPath": "background.overlay"
		},
		{
			"cssVar": "text-primary",
			"tokenPath": "text.primary"
		},
		{
			"cssVar": "text-secondary",
			"tokenPath": "text.secondary"
		},
		{
			"cssVar": "text-inverse",
			"tokenPath": "text.inverse"
		},
		{
			"cssVar": "brand-primary",
			"tokenPath": "brand.primary"
		},
		{
			"cssVar": "state-info",
			"tokenPath": "state.info"
		},
		{
			"cssVar": "state-success",
			"tokenPath": "state.success"
		},
		{
			"cssVar": "state-warning",
			"tokenPath": "state.warning"
		},
		{
			"cssVar": "state-danger",
			"tokenPath": "state.danger"
		},
		{
			"cssVar": "border-default",
			"tokenPath": "border.default"
		},
		{
			"cssVar": "accent-link",
			"tokenPath": "accent.link"
		},
		{
			"cssVar": "font-sans",
			"tokenPath": "typography.fonts.sans"
		},
		{
			"cssVar": "font-mono",
			"tokenPath": "typography.fonts.mono"
		},
		{
			"cssVar": "heading-h1",
			"tokenPath": "content.heading.h1"
		},
		{
			"cssVar": "heading-h2",
			"tokenPath": "content.heading.h2"
		},
		{
			"cssVar": "heading-h3",
			"tokenPath": "content.heading.h3"
		},
		{
			"cssVar": "heading-h4",
			"tokenPath": "content.heading.h4"
		},
		{
			"cssVar": "heading-h5",
			"tokenPath": "content.heading.h5"
		},
		{
			"cssVar": "heading-h6",
			"tokenPath": "content.heading.h6"
		},
		{
			"cssVar": "body-primary",
			"tokenPath": "content.body.primary"
		},
		{
			"cssVar": "body-secondary",
			"tokenPath": "content.body.secondary"
		},
		{
			"cssVar": "link-default",
			"tokenPath": "content.link.default"
		},
		{
			"cssVar": "selection-fg",
			"tokenPath": "content.selection.fg"
		},
		{
			"cssVar": "selection-bg",
			"tokenPath": "content.selection.bg"
		},
		{
			"cssVar": "blockquote-border",
			"tokenPath": "content.blockquote.border"
		},
		{
			"cssVar": "blockquote-fg",
			"tokenPath": "content.blockquote.fg"
		},
		{
			"cssVar": "blockquote-bg",
			"tokenPath": "content.blockquote.bg"
		},
		{
			"cssVar": "code-inline-fg",
			"tokenPath": "content.codeInline.fg"
		},
		{
			"cssVar": "code-inline-bg",
			"tokenPath": "content.codeInline.bg"
		},
		{
			"cssVar": "code-block-fg",
			"tokenPath": "content.codeBlock.fg"
		},
		{
			"cssVar": "code-block-bg",
			"tokenPath": "content.codeBlock.bg"
		},
		{
			"cssVar": "table-border",
			"tokenPath": "content.table.border"
		},
		{
			"cssVar": "table-stripe",
			"tokenPath": "content.table.stripe"
		},
		{
			"cssVar": "table-thead-bg",
			"tokenPath": "content.table.theadBg"
		},
		{
			"cssVar": "table-cell-bg",
			"tokenPath": "content.table.cellBg",
			"fallback": "background.base"
		},
		{
			"cssVar": "table-header-fg",
			"tokenPath": "content.table.headerFg",
			"fallback": "text.primary"
		}
	],
	optionalGroups: {
		"spacing": {
			"prefix": "spacing",
			"properties": [
				"xs",
				"sm",
				"md",
				"lg",
				"xl"
			]
		},
		"elevation": {
			"prefix": "elevation",
			"properties": [
				"none",
				"sm",
				"md",
				"lg",
				"xl"
			]
		},
		"animation": {
			"prefix": "animation",
			"mappings": [
				{
					"cssVar": "duration-fast",
					"tokenPath": "animation.durationFast"
				},
				{
					"cssVar": "duration-normal",
					"tokenPath": "animation.durationNormal"
				},
				{
					"cssVar": "duration-slow",
					"tokenPath": "animation.durationSlow"
				},
				{
					"cssVar": "easing-default",
					"tokenPath": "animation.easingDefault"
				},
				{
					"cssVar": "easing-emphasized",
					"tokenPath": "animation.easingEmphasized"
				}
			]
		},
		"opacity": {
			"prefix": "opacity",
			"properties": [
				"disabled",
				"hover",
				"pressed"
			]
		}
	},
	bulma: {
		"schemeMappings": [
			{
				"cssVar": "--bulma-body-background-color",
				"tokenPath": "background.base"
			},
			{
				"cssVar": "--bulma-body-color",
				"tokenPath": "text.primary"
			},
			{
				"cssVar": "--bulma-scheme-main",
				"tokenPath": "background.base"
			},
			{
				"cssVar": "--bulma-scheme-main-bis",
				"tokenPath": "background.surface"
			},
			{
				"cssVar": "--bulma-scheme-main-ter",
				"tokenPath": "background.overlay"
			},
			{
				"cssVar": "--bulma-scheme-invert",
				"tokenPath": "text.primary"
			},
			{
				"cssVar": "--bulma-border",
				"tokenPath": "border.default"
			}
		],
		"primaryHsl": {
			"hVar": "--bulma-primary-h",
			"sVar": "--bulma-primary-s",
			"lVar": "--bulma-primary-l",
			"tokenPath": "brand.primary"
		},
		"stateHslMappings": [
			{
				"cssVar": "--bulma-link",
				"tokenPath": "accent.link"
			},
			{
				"cssVar": "--bulma-info",
				"tokenPath": "state.info"
			},
			{
				"cssVar": "--bulma-success",
				"tokenPath": "state.success"
			},
			{
				"cssVar": "--bulma-warning",
				"tokenPath": "state.warning"
			},
			{
				"cssVar": "--bulma-danger",
				"tokenPath": "state.danger"
			}
		]
	},
	theme: {
		"textMappings": [{
			"cssVar": "--theme-text",
			"tokenPath": "text.primary"
		}, {
			"cssVar": "--theme-text-muted",
			"tokenPath": "text.secondary"
		}],
		"headingMappings": [
			{
				"cssVar": "--theme-h1",
				"tokenPath": "content.heading.h1"
			},
			{
				"cssVar": "--theme-h2",
				"tokenPath": "content.heading.h2"
			},
			{
				"cssVar": "--theme-h3",
				"tokenPath": "content.heading.h3"
			},
			{
				"cssVar": "--theme-h4",
				"tokenPath": "content.heading.h4"
			},
			{
				"cssVar": "--theme-h5",
				"tokenPath": "content.heading.h5"
			},
			{
				"cssVar": "--theme-h6",
				"tokenPath": "content.heading.h6"
			}
		],
		"contentMappings": [
			{
				"cssVar": "--theme-link",
				"tokenPath": "accent.link"
			},
			{
				"cssVar": "--theme-blockquote-border",
				"tokenPath": "content.blockquote.border"
			},
			{
				"cssVar": "--theme-blockquote-fg",
				"tokenPath": "content.blockquote.fg"
			},
			{
				"cssVar": "--theme-blockquote-bg",
				"tokenPath": "content.blockquote.bg"
			},
			{
				"cssVar": "--theme-code-fg",
				"tokenPath": "content.codeInline.fg"
			},
			{
				"cssVar": "--theme-code-bg",
				"tokenPath": "content.codeInline.bg"
			},
			{
				"cssVar": "--theme-pre-fg",
				"tokenPath": "content.codeBlock.fg"
			},
			{
				"cssVar": "--theme-pre-bg",
				"tokenPath": "content.codeBlock.bg"
			},
			{
				"cssVar": "--theme-selection-fg",
				"tokenPath": "content.selection.fg"
			},
			{
				"cssVar": "--theme-selection-bg",
				"tokenPath": "content.selection.bg"
			}
		],
		"tableMappings": [
			{
				"cssVar": "--theme-table-border",
				"tokenPath": "content.table.border"
			},
			{
				"cssVar": "--theme-table-stripe",
				"tokenPath": "content.table.stripe"
			},
			{
				"cssVar": "--theme-table-thead-bg",
				"tokenPath": "content.table.theadBg"
			},
			{
				"cssVar": "--theme-table-cell-bg",
				"tokenPath": "content.table.cellBg",
				"fallback": "background.base"
			},
			{
				"cssVar": "--theme-table-header-fg",
				"tokenPath": "content.table.headerFg",
				"fallback": "text.primary"
			}
		],
		"syntaxMappings": [
			{
				"cssVar": "--syntax-fg",
				"tokenPath": "content.codeInline.fg"
			},
			{
				"cssVar": "--syntax-bg",
				"tokenPath": "content.codeInline.bg"
			},
			{
				"cssVar": "--syntax-keyword",
				"tokenPath": "brand.primary"
			},
			{
				"cssVar": "--syntax-string",
				"tokenPath": "state.success"
			},
			{
				"cssVar": "--syntax-number",
				"tokenPath": "state.warning"
			},
			{
				"cssVar": "--syntax-comment",
				"tokenPath": "text.secondary"
			},
			{
				"cssVar": "--syntax-title",
				"tokenPath": "state.info"
			},
			{
				"cssVar": "--syntax-attr",
				"tokenPath": "accent.link"
			}
		],
		"surfaceMappings": [
			{
				"cssVar": "--theme-surface-0",
				"tokenPath": "background.base"
			},
			{
				"cssVar": "--theme-surface-1",
				"tokenPath": "background.surface"
			},
			{
				"cssVar": "--theme-surface-2",
				"tokenPath": "background.overlay"
			},
			{
				"cssVar": "--theme-border",
				"tokenPath": "border.default"
			}
		]
	},
	components: [
		{
			"cssVar": "--theme-card-bg",
			"componentPath": "card.bg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-card-border",
			"componentPath": "card.border",
			"fallback": "border.default"
		},
		{
			"cssVar": "--theme-card-header-bg",
			"componentPath": "card.headerBg",
			"fallback": "background.overlay"
		},
		{
			"cssVar": "--theme-card-footer-bg",
			"componentPath": "card.footerBg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-message-bg",
			"componentPath": "message.bg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-message-header-bg",
			"componentPath": "message.headerBg",
			"fallback": "background.overlay"
		},
		{
			"cssVar": "--theme-message-border",
			"componentPath": "message.border",
			"fallback": "border.default"
		},
		{
			"cssVar": "--theme-message-body-fg",
			"componentPath": "message.bodyFg",
			"fallback": "text.primary"
		},
		{
			"cssVar": "--theme-panel-bg",
			"componentPath": "panel.bg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-panel-header-bg",
			"componentPath": "panel.headerBg",
			"fallback": "background.overlay"
		},
		{
			"cssVar": "--theme-panel-header-fg",
			"componentPath": "panel.headerFg",
			"fallback": "text.primary"
		},
		{
			"cssVar": "--theme-panel-border",
			"componentPath": "panel.border",
			"fallback": "border.default"
		},
		{
			"cssVar": "--theme-panel-block-bg",
			"componentPath": "panel.blockBg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-panel-block-hover-bg",
			"componentPath": "panel.blockHoverBg",
			"fallback": "background.overlay"
		},
		{
			"cssVar": "--theme-panel-block-active-bg",
			"componentPath": "panel.blockActiveBg",
			"fallback": "background.overlay"
		},
		{
			"cssVar": "--theme-box-bg",
			"componentPath": "box.bg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-box-border",
			"componentPath": "box.border",
			"fallback": "border.default"
		},
		{
			"cssVar": "--theme-notification-bg",
			"componentPath": "notification.bg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-notification-border",
			"componentPath": "notification.border",
			"fallback": "border.default"
		},
		{
			"cssVar": "--theme-modal-bg",
			"componentPath": "modal.bg",
			"fallback": ""
		},
		{
			"cssVar": "--theme-modal-card-bg",
			"componentPath": "modal.cardBg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-modal-header-bg",
			"componentPath": "modal.headerBg",
			"fallback": "background.overlay"
		},
		{
			"cssVar": "--theme-modal-footer-bg",
			"componentPath": "modal.footerBg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-dropdown-bg",
			"componentPath": "dropdown.bg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-dropdown-item-hover-bg",
			"componentPath": "dropdown.itemHoverBg",
			"fallback": "background.overlay"
		},
		{
			"cssVar": "--theme-dropdown-border",
			"componentPath": "dropdown.border",
			"fallback": "border.default"
		},
		{
			"cssVar": "--theme-tabs-border",
			"componentPath": "tabs.border",
			"fallback": "border.default"
		},
		{
			"cssVar": "--theme-tabs-link-bg",
			"componentPath": "tabs.linkBg",
			"fallback": "background.surface"
		},
		{
			"cssVar": "--theme-tabs-link-active-bg",
			"componentPath": "tabs.linkActiveBg",
			"fallback": "background.base"
		},
		{
			"cssVar": "--theme-tabs-link-hover-bg",
			"componentPath": "tabs.linkHoverBg",
			"fallback": "background.overlay"
		}
	],
	syntaxClassGroups: {
		"comment": [
			".c",
			".cm",
			".c1"
		],
		"keyword": [
			".k",
			".kc",
			".kd"
		],
		"string": [
			".s",
			".s1",
			".sb",
			".sd",
			".s2"
		],
		"number": [
			".m",
			".mi",
			".mf",
			".mh"
		],
		"attr": [".nt", ".na"],
		"title": [
			".nn",
			".nc",
			".no",
			".nf"
		]
	}
};
//#endregion
//#region ../core/dist/themes/css/mappings.js
function toVariableMapping(entry) {
	return {
		cssVar: entry.cssVar,
		tokenPath: entry.tokenPath,
		...entry.fallback ? { fallbackPath: entry.fallback } : {}
	};
}
function toComponentMapping(entry) {
	return {
		cssVar: entry.cssVar,
		componentPath: entry.componentPath,
		fallbackPath: entry.fallback
	};
}
function toHSLMapping(entry) {
	return {
		hVar: entry.hVar,
		sVar: entry.sVar,
		lVar: entry.lVar,
		tokenPath: entry.tokenPath
	};
}
function toHSLColorMapping(entry) {
	return {
		cssVar: entry.cssVar,
		tokenPath: entry.tokenPath
	};
}
token_mappings_default.bulma.schemeMappings.map(toVariableMapping);
toHSLMapping(token_mappings_default.bulma.primaryHsl);
token_mappings_default.bulma.stateHslMappings.map(toHSLColorMapping);
token_mappings_default.theme.textMappings.map(toVariableMapping);
token_mappings_default.theme.headingMappings.map(toVariableMapping);
token_mappings_default.theme.contentMappings.map(toVariableMapping);
token_mappings_default.theme.tableMappings.map(toVariableMapping);
token_mappings_default.theme.syntaxMappings.map(toVariableMapping);
token_mappings_default.theme.surfaceMappings.map(toVariableMapping);
token_mappings_default.components.map(toComponentMapping);
token_mappings_default.syntaxClassGroups;
var CORE_MAPPINGS = token_mappings_default.coreMappings.map(toVariableMapping);
var CSS_VAR_PREFIX = token_mappings_default.prefix;
var OPTIONAL_GROUPS = token_mappings_default.optionalGroups;
//#endregion
//#region src/generator.ts
function resolveTokenPath(tokens, path) {
	const parts = path.split(".");
	let current = tokens;
	for (const part of parts) {
		if (current === null || current === void 0 || typeof current !== "object") return;
		current = current[part];
	}
	return typeof current === "string" ? current : void 0;
}
var DEFAULT_MODAL_BACKDROP = "rgba(10, 10, 10, 0.86)";
var TRUSTED_FONT_DOMAINS = [
	"fonts.googleapis.com",
	"fonts.gstatic.com",
	"use.typekit.net",
	"fonts.bunny.net",
	"rsms.me",
	"fonts.cdnfonts.com",
	"github.githubassets.com"
];
function isValidFontUrl(url) {
	try {
		const parsed = new URL(url);
		if (parsed.protocol !== "https:") return false;
		return TRUSTED_FONT_DOMAINS.some((domain) => parsed.hostname === domain || parsed.hostname.endsWith(`.${domain}`));
	} catch {
		return false;
	}
}
var COMPONENT_TOKEN_MAPPINGS = [
	[
		"card-bg",
		"card.bg",
		"background.surface"
	],
	[
		"card-border",
		"card.border",
		"border.default"
	],
	[
		"card-header-bg",
		"card.headerBg",
		"background.overlay"
	],
	[
		"card-footer-bg",
		"card.footerBg",
		"background.surface"
	],
	[
		"message-bg",
		"message.bg",
		"background.surface"
	],
	[
		"message-header-bg",
		"message.headerBg",
		"background.overlay"
	],
	[
		"message-border",
		"message.border",
		"border.default"
	],
	[
		"message-body-fg",
		"message.bodyFg",
		"text.primary"
	],
	[
		"panel-bg",
		"panel.bg",
		"background.surface"
	],
	[
		"panel-header-bg",
		"panel.headerBg",
		"background.overlay"
	],
	[
		"panel-header-fg",
		"panel.headerFg",
		"text.primary"
	],
	[
		"panel-border",
		"panel.border",
		"border.default"
	],
	[
		"panel-block-bg",
		"panel.blockBg",
		"background.surface"
	],
	[
		"panel-block-hover-bg",
		"panel.blockHoverBg",
		"background.overlay"
	],
	[
		"panel-block-active-bg",
		"panel.blockActiveBg",
		"background.overlay"
	],
	[
		"box-bg",
		"box.bg",
		"background.surface"
	],
	[
		"box-border",
		"box.border",
		"border.default"
	],
	[
		"notification-bg",
		"notification.bg",
		"background.surface"
	],
	[
		"notification-border",
		"notification.border",
		"border.default"
	],
	[
		"modal-bg",
		"modal.bg",
		""
	],
	[
		"modal-card-bg",
		"modal.cardBg",
		"background.surface"
	],
	[
		"modal-header-bg",
		"modal.headerBg",
		"background.overlay"
	],
	[
		"modal-footer-bg",
		"modal.footerBg",
		"background.surface"
	],
	[
		"dropdown-bg",
		"dropdown.bg",
		"background.surface"
	],
	[
		"dropdown-item-hover",
		"dropdown.itemHoverBg",
		"background.overlay"
	],
	[
		"dropdown-border",
		"dropdown.border",
		"border.default"
	],
	[
		"tabs-border",
		"tabs.border",
		"border.default"
	],
	[
		"tabs-link-bg",
		"tabs.linkBg",
		"background.surface"
	],
	[
		"tabs-link-active-bg",
		"tabs.linkActiveBg",
		"background.base"
	],
	[
		"tabs-link-hover-bg",
		"tabs.linkHoverBg",
		"background.overlay"
	]
];
function resolveComponentToken(tokens, componentPath, fallbackPath) {
	const componentValue = resolveTokenPath(tokens, `components.${componentPath}`);
	if (componentValue !== void 0) return componentValue;
	if (componentPath === "modal.bg") return DEFAULT_MODAL_BACKDROP;
	if (fallbackPath) return resolveTokenPath(tokens, fallbackPath) ?? "";
	return "";
}
function generateCssVarsFromTokens(tokens) {
	const lines = [];
	const prefix = CSS_VAR_PREFIX;
	const add = (name, value) => {
		lines.push(`  --${prefix}-${name}: ${value};`);
	};
	for (const mapping of CORE_MAPPINGS) {
		const varName = mapping.cssVar.startsWith("--") ? mapping.cssVar.slice(2) : mapping.cssVar;
		let value = resolveTokenPath(tokens, mapping.tokenPath);
		if (value === void 0 && mapping.fallbackPath) value = resolveTokenPath(tokens, mapping.fallbackPath);
		if (value !== void 0) add(varName, value);
	}
	const optionalGroups = OPTIONAL_GROUPS;
	if (tokens.spacing && optionalGroups.spacing) {
		const { properties } = optionalGroups.spacing;
		for (const prop of properties) {
			const value = tokens.spacing[prop];
			if (value) add(`${optionalGroups.spacing.prefix}-${prop}`, value);
		}
	}
	if (tokens.elevation && optionalGroups.elevation) {
		const { properties } = optionalGroups.elevation;
		for (const prop of properties) {
			const value = tokens.elevation[prop];
			if (value) add(`${optionalGroups.elevation.prefix}-${prop}`, value);
		}
	}
	if (tokens.animation && optionalGroups.animation?.mappings) for (const mapping of optionalGroups.animation.mappings) {
		const value = resolveTokenPath(tokens, mapping.tokenPath);
		if (value) add(`${optionalGroups.animation.prefix}-${mapping.cssVar}`, value);
	}
	if (tokens.opacity && optionalGroups.opacity) {
		const { properties } = optionalGroups.opacity;
		for (const prop of properties) {
			const value = tokens.opacity[prop];
			if (value !== void 0) add(`${optionalGroups.opacity.prefix}-${prop}`, String(value));
		}
	}
	for (const [cssVar, componentPath, fallbackPath] of COMPONENT_TOKEN_MAPPINGS) {
		const value = resolveComponentToken(tokens, componentPath, fallbackPath);
		if (value) add(cssVar, value);
	}
	const syntaxLines = generateSyntaxVarsFromTokens(tokens);
	lines.push(...syntaxLines);
	return lines;
}
function generateThemeCss(flavor) {
	const vars = generateCssVarsFromTokens(flavor.tokens);
	const fontImports = (flavor.tokens.typography?.webFonts ?? []).filter(isValidFontUrl).map((url) => `@import url('${url}');`).join("\n");
	const colorScheme = flavor.appearance === "dark" ? "dark" : "light";
	const cssContent = `[data-theme="${flavor.id}"] {
${vars.join("\n")}
  color-scheme: ${colorScheme};
}`;
	return fontImports ? `${fontImports}\n\n${cssContent}\n` : `${cssContent}\n`;
}
var DESIGN_SYSTEM_TOKENS = `
  /* Spacing */
  --space-xs: 0.25rem;
  --space-sm: 0.5rem;
  --space-md: 1rem;
  --space-lg: 1.5rem;
  --space-xl: 2rem;
  --space-2xl: 3rem;
  --space-3xl: 4rem;
  --space-4xl: 6rem;

  /* Border Radius */
  --radius-sm: 0.5rem;
  --radius-md: 0.75rem;
  --radius-lg: 1rem;
  --radius-xl: 1.5rem;
  --radius-2xl: 2rem;
  --radius-full: 9999px;

  /* Shadows */
  --shadow-sm: 0 1px 2px rgba(0, 0, 0, 0.05);
  --shadow-md: 0 4px 6px rgba(0, 0, 0, 0.07), 0 2px 4px rgba(0, 0, 0, 0.05);
  --shadow-lg: 0 10px 15px rgba(0, 0, 0, 0.1), 0 4px 6px rgba(0, 0, 0, 0.05);
  --shadow-xl: 0 20px 25px rgba(0, 0, 0, 0.15), 0 10px 10px rgba(0, 0, 0, 0.04);
  --shadow-glow: 0 0 30px color-mix(in srgb, var(--turbo-brand-primary) 30%, transparent);
  --shadow-glow-sm: 0 0 15px color-mix(in srgb, var(--turbo-brand-primary) 20%, transparent);

  /* Transitions */
  --transition-fast: 120ms ease-out;
  --transition-normal: 200ms ease-out;
  --transition-slow: 300ms cubic-bezier(0.4, 0, 0.2, 1);

  /* Gradients (theme-aware) */
  --gradient-primary: linear-gradient(135deg, var(--turbo-brand-primary), var(--turbo-state-info));
  --gradient-surface: linear-gradient(180deg, var(--turbo-bg-surface), var(--turbo-bg-base));
`;
function generateCoreCss(defaultFlavor) {
	return `:root {\n${generateCssVarsFromTokens(defaultFlavor.tokens).join("\n")}\n${DESIGN_SYSTEM_TOKENS}}\n`;
}
function generateCombinedCss(flavors, defaultFlavorId = "catppuccin-mocha") {
	const defaultFlavor = flavors.find((f) => f.id === defaultFlavorId) || flavors[0];
	if (!defaultFlavor) throw new Error("No flavors provided");
	return `/* Turbo Themes - Pure CSS Custom Properties */\n/* Generated automatically - do not edit */\n\n${generateCoreCss(defaultFlavor)}\n${flavors.map((flavor) => generateThemeCss(flavor)).join("\n")}`;
}
function generateThemesOnlyCss(flavors) {
	if (!flavors || flavors.length === 0) throw new Error("No flavors provided");
	const allImports = [];
	const themeBlocks = [];
	for (const flavor of flavors) {
		const lines = generateThemeCss(flavor).split("\n");
		const imports = [];
		const rest = [];
		for (const line of lines) if (line.startsWith("@import ")) imports.push(line);
		else rest.push(line);
		for (const imp of imports) if (!allImports.includes(imp)) allImports.push(imp);
		themeBlocks.push(rest.join("\n"));
	}
	return `${allImports.length > 0 ? `${allImports.join("\n")}\n\n` : ""}/* Turbo Themes - All Theme Selectors (no :root defaults) */
/* Load alongside turbo-core.css + turbo-base.css for FOUC-free theme switching */
/* Generated automatically - do not edit */\n\n${themeBlocks.join("\n")}`;
}
//#endregion
//#region src/base.ts
var CSS_RESET = `/* Turbo Themes - Minimal Reset */
*,
*::before,
*::after {
  box-sizing: border-box;
}

body,
h1,
h2,
h3,
h4,
h5,
h6,
p,
figure,
blockquote,
dl,
dd {
  margin: 0;
}

html {
  -webkit-text-size-adjust: 100%;
  -moz-text-size-adjust: 100%;
  text-size-adjust: 100%;
}

body {
  min-height: 100vh;
  line-height: 1.5;
  -webkit-font-smoothing: antialiased;
}

img,
picture,
video,
canvas,
svg {
  display: block;
  max-width: 100%;
}

input,
button,
textarea,
select {
  font: inherit;
}
`;
var CSS_BASE = `/* Turbo Themes - Base Semantic Styles */

/* Root & Body */
:root {
  color-scheme: light dark;
}

body {
  font-family: var(--turbo-font-sans, system-ui, -apple-system, sans-serif);
  color: var(--turbo-text-primary, #000);
  background-color: var(--turbo-bg-base, #fff);
}

/* Typography */
h1, h2, h3, h4, h5, h6 {
  line-height: 1.2;
  font-weight: 600;
}

h1 { color: var(--turbo-heading-h1, var(--turbo-text-primary)); font-size: 2.5rem; }
h2 { color: var(--turbo-heading-h2, var(--turbo-text-primary)); font-size: 2rem; }
h3 { color: var(--turbo-heading-h3, var(--turbo-text-primary)); font-size: 1.75rem; }
h4 { color: var(--turbo-heading-h4, var(--turbo-text-primary)); font-size: 1.5rem; }
h5 { color: var(--turbo-heading-h5, var(--turbo-text-primary)); font-size: 1.25rem; }
h6 { color: var(--turbo-heading-h6, var(--turbo-text-primary)); font-size: 1rem; }

p {
  color: var(--turbo-body-primary, var(--turbo-text-primary));
}

/* Links */
a {
  color: var(--turbo-accent-link, var(--turbo-brand-primary));
  text-decoration: none;
}

a:hover {
  text-decoration: underline;
}

/* Selection */
::selection {
  color: var(--turbo-selection-fg, #fff);
  background-color: var(--turbo-selection-bg, #3273dc);
}

/* Code */
code,
kbd,
samp,
pre {
  font-family: var(--turbo-font-mono, ui-monospace, monospace);
}

code {
  padding: 0.125rem 0.25rem;
  color: var(--turbo-code-inline-fg, #e83e8c);
  background-color: var(--turbo-code-inline-bg, #f8f9fa);
  border-radius: 0.25rem;
  font-size: 0.875em;
}

pre {
  padding: 1rem;
  color: var(--turbo-code-block-fg, #f8f9fa);
  background-color: var(--turbo-code-block-bg, #2b2b2b);
  border-radius: 0.5rem;
  overflow-x: auto;
}

pre code {
  padding: 0;
  background-color: transparent;
  font-size: inherit;
}

/* Blockquote */
blockquote {
  margin: 1rem 0;
  padding: 0.5rem 1rem;
  color: var(--turbo-blockquote-fg);
  background-color: var(--turbo-blockquote-bg);
  border-left: 4px solid var(--turbo-blockquote-border);
  border-radius: 0 0.25rem 0.25rem 0;
}

/* Tables */
table {
  width: 100%;
  border-collapse: collapse;
}

th,
td {
  padding: 0.75rem;
  text-align: left;
  border: 1px solid var(--turbo-table-border);
}

th {
  color: var(--turbo-table-header-fg, var(--turbo-text-primary));
  background-color: var(--turbo-table-thead-bg);
  font-weight: 600;
}

tr:nth-child(even) td {
  background-color: var(--turbo-table-stripe);
}

/* Form Elements */
button,
input,
select,
textarea {
  border: 1px solid var(--turbo-border-default);
  border-radius: 0.25rem;
  background-color: var(--turbo-bg-surface);
  color: var(--turbo-text-primary);
}

button {
  padding: 0.5rem 1rem;
  cursor: pointer;
  background-color: var(--turbo-brand-primary);
  color: var(--turbo-text-inverse);
  border: none;
  font-weight: 500;
}

button:hover {
  opacity: 0.9;
}

button:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

input,
select,
textarea {
  padding: 0.5rem 0.75rem;
}

button:focus-visible,
input:focus,
select:focus,
textarea:focus {
  outline: 2px solid var(--turbo-brand-primary);
  outline-offset: 2px;
}

/* Utility classes */
.surface {
  background-color: var(--turbo-bg-surface);
}

.surface-alt {
  background-color: var(--turbo-bg-overlay);
}

.text-primary {
  color: var(--turbo-text-primary);
}

.text-secondary {
  color: var(--turbo-text-secondary);
}

.text-success {
  color: var(--turbo-state-success);
}

.text-warning {
  color: var(--turbo-state-warning);
}

.text-danger {
  color: var(--turbo-state-danger);
}

.text-info {
  color: var(--turbo-state-info);
}

.border {
  border: 1px solid var(--turbo-border-default);
}

.rounded {
  border-radius: 0.25rem;
}

.rounded-lg {
  border-radius: 0.5rem;
}
`;
function generateBaseCss() {
	return `${CSS_RESET}\n${CSS_BASE}`;
}
function generateResetCss() {
	return CSS_RESET;
}
//#endregion
export { CSS_BASE, CSS_RESET, CSS_SYNTAX, CSS_VAR_PREFIX, generateBaseCss, generateCombinedCss, generateCoreCss, generateCssVarsFromTokens, generateResetCss, generateSyntaxBaseCss, generateSyntaxCss, generateSyntaxVarsFromTokens, generateThemeCss, generateThemesOnlyCss };
